src/Features/CSharpTest/ImplementAbstractClass/ImplementAbstractClassTests_ThroughMember.cs

```diff
@@ -826,5 +826,107 @@ public override void Method1()
                 </Workspace>
                 """, index: 1);
         }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/69177")]
+        public async Task TestImplementThroughPrimaryConstructorParam1()
+        {
+            await TestInRegularAndScriptAsync(
+                """
+                abstract class Base
+                {
+                    public abstract int Method();
+                }
+
+                class [|Program|](Base base1) : Base
+                {
+                }
+                """,
+                """
+                abstract class Base
+                {
+                    public abstract int Method();
+                }
+
+                class Program(Base base1) : Base
+                {
+                    public override int Method()
+                    {
+                        return base1.Method();
+                    }
+                }
+                """, index: 1);
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/69177")]
+        public async Task TestImplementThroughPrimaryConstructorParam2()
+        {
+            // Don't offer "implement through" since this PC parameter is captured as a field.
```

Comment:
💡 I believe the refactoring is technically offered, just by using the field instead of the primary constructor parameter